src/map/script.c

```diff
@@ -26510,6 +26523,18 @@ static BUILDIN(rodex_sendmail)
 			return false;
 		}
 
+		if (!data_isint(script_getdata(st, param + 1))) {
+			ShowError("script:rodex_sendmail: Passed amount for item %d is not a number!\n", i + 1);
+			return false;
+		}
+
+		int amount = script_getnum(st, param + 1);
+
+		if (amount < 1 || amount > min(MAX_AMOUNT, SHRT_MAX)) {
```

Comment:
are you sure about using `min(MAX_AMOUNT, SHRT_MAX)` here ?
if let's say gravity decided to pump up the MAX_AMOUNT to INT_MAX

--> NOTE gravity already pump up their MAX_ITEM_ID from short into int

then it will become SHRT_MAX instead of actual MAX_AMOUNT ...
MAX_AMOUNT is define in _src\common\mmo.h_ for this reason

Reply:
`msg.items[i].item.amount` is of type `short` and thus this limitation if correct for now.

Reply:
so in the future if gravity change this into INT ...
they have to change this line too ?

Reply:
**IF** Gravity one day will change this (which I strongly doubt) we'd have to change the data type of field `amount` in `struct item` (mmo.h) and field `count` in `struct PACKET_CZ_ADD_ITEM_TO_MAIL`/`struct PACKET_ZC_ADD_ITEM_TO_MAIL` (packets_struct.h).